compiler/rustc_mir_build/src/errors.rs

```diff
@@ -453,14 +453,14 @@ pub enum UnusedUnsafeEnclosing {
     },
 }
 
-pub(crate) struct NonExhaustivePatternsTypeNotEmpty<'p, 'tcx, 'm> {
-    pub cx: &'m RustcMatchCheckCtxt<'p, 'tcx>,
+pub(crate) struct NonExhaustivePatternsTypeNotEmpty<'p, 'thir, 'tcx, 'm> {
```

Comment:
When is `'p` shorter than `'thir`? Isn't `'p` the lifetime of the "pattern"? Shouldn't that be from the THIR anyways?

Reply:
Hehe, my solution to lifetime problems tends to be "add more lifetimes".

`'thir` is the lifetime of `thir::Pat` patterns we indeed get from the thir. `'p` is the lifetime of the `DeconstructedPat`s we allocate in an arena. I agree that it feels like `'p` should be all we need, lemme try that again now that I understand the situation better

Reply:
If `'thir` and `'p` are both covariant, then they should be able to be unified. But maybe not -- I guess give it a shot but r=me if it doesn't end up working out.